fix-android-dependencies.py

```diff
@@ -0,0 +1,109 @@
+import json
+import os
+import re
+
+# These will have to be updated manually over time, there's not an
+# easy way to determine the latest version.
+COMPILE_SDK_VERSION = 27
+TARGET_SDK_VERSION = 27
+BUILD_TOOLS_VERSION = '27.0.2'
+
+COMPILE_SDK_RE = r'compileSdkVersion[\s][\w]+'
+TARGET_SDK_RE = r'targetSdkVersion[\s][\w]+'
+BUILD_TOOLS_RE = r'buildToolsVersion[\s][\'\"][\w\.]+[\'\"]'
+
+GROUPS_ALLOWED_MAJOR_UPDATE = [
+  'com.google.android.gms',
+  'com.android.support'
+]
+
+# Depends on https://github.com/ben-manes/gradle-versions-plugin
+#
+# Must run this command:
+# $ ./gradlew dependencyUpdates -Drevision=release -DoutputFormatter=json
+INPUT_JSON = 'build/dependencyUpdates/report.json'
+
+def findGradleFiles():
+    """Finds all build.gradle files, recursively."""
+    gradle_files = []
+    for root, dirs, files in os.walk('.'):
+        for file in files:
+            if file.endswith('build.gradle'):
+                gradle_files.append(os.path.join(root, file))
+
+    return gradle_files
+
+def getAndroidReplacements():
+    """Gets a dictionary of all android-specific replacements to be made."""
+    replacements = {}
+
+    compileSdk = 'compileSdkVersion {}'.format(COMPILE_SDK_VERSION)
+    targetSdk = 'targetSdkVersion {}'.format(TARGET_SDK_VERSION)
+    buildToolsVersion = 'buildToolsVersion \'{}\''.format(BUILD_TOOLS_VERSION)
+
+    replacements[COMPILE_SDK_RE] = compileSdk
+    replacements[TARGET_SDK_RE] = targetSdk
+    replacements[BUILD_TOOLS_RE] = buildToolsVersion
+
+    return replacements
+
+def isMajorUpdate(old_version, new_version):
+  """Compares version strings to see if it's a major update."""
+  old_major = old_version.split('.')[0]
```

Comment:
this one function seems to use 2 space tabs, please use 4 space tabs throughout.